arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/LowGasStipendPrecompileCallTests.java

```diff
@@ -0,0 +1,260 @@
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.zktracer.module.hub.Hub;
+import net.consensys.linea.zktracer.module.oob.OobOperation;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class LowGasStipendPrecompileCallTests {
+
+  // Enums for the different testing scenarios
+  enum ValueParameter {
+    ZERO,
+    NON_ZERO;
+
+    boolean isZeroArgument() {
+      return this == ZERO;
+    }
+
+    boolean isNonZeroArgument() {
+      return this == NON_ZERO;
+    }
+  }
+
+  enum GasParameter {
+    ZERO,
+    ONE,
+    COST_MINUS_ONE,
+    COST,
+    COST_PLUS_ONE;
+  }
+
+  @ParameterizedTest
+  @MethodSource("lowGasStipendPrecompileCallTestSource")
+  void lowGasStipendPrecompileCallTest(
+      Address precompileAddress, ValueParameter valueParameter, GasParameter gasParameter) {
+    final BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    // In order to actually trigger the insufficient we need to:
+    // - Set a specific args size for BLAKE2F and EC_PAIRING
+    // - Set the r value of BLAKE2F to have precompileCost > gasBonus
+    // - Populate the memory with a large enough number of words for SHA256, RIPEMD160, and ID
+    //   to have precompileCost > gasBonus.
+    final int value = valueParameter.isZeroArgument() ? 0 : 1;
+    final int argsSize; // depends on the called precompile
+    final int argsOffset = valueParameter.isZeroArgument() ? 0 : 1;
+    final int retSize = valueParameter.isZeroArgument() ? 0 : 1;
+    final int retOffset = valueParameter.isZeroArgument() ? 0 : 1;
+
+    // BLAKE2F specific parameters
+    int rFirstByte = valueParameter.isZeroArgument() ? 0 : 0x12;
+    int r = rFirstByte << 8;
+    if (precompileAddress == BLAKE2B_F_COMPRESSION) {
+      program
+          .push(rFirstByte) // For simplicity, we only set the first byte of r
+          .push(argsOffset + 2) // offset
+          // Writing rFirstByte at this offset
+          // allows to have r = 0x00000000 or r = 0x00001200
+          .op(OpCode.MSTORE8);
+      argsSize = 213;
+    } else if (precompileAddress == ALTBN128_PAIRING) {
+      // EC_PAIRING specific parameters
+      argsSize = 192;
+    } else if ((precompileAddress == SHA256
+            || precompileAddress == RIPEMD160
+            || precompileAddress == ID)
+        && valueParameter.isNonZeroArgument()) {
+      // SHA256, RIPEMD160, and ID specific parameters
+      int nWords = 1024;
+      argsSize = nWords * WORD_SIZE; // This guarantees that precompileCost > gasBonus
```

Comment:
The spec calls it `G_callstipend`, we should thus call it `callStipend` rather than `gasBonus`.